added category[most,least,neutral] validation

Pratik-7331-sjsu opened the pull request:
This PR adds validation for the upsertwhy category to ensure that only valid inputs (least, most, neutral) are accepted and invalid or empty inputs are handled correctly.

Implemented validation using Joi to check for valid, empty, or undefined categories.
Added Jest tests to cover validation logic.
Updated code comments to reflect the changes.

ddfridley reviewed (changes requested):
This is looking good. Can the change so packackage.json and package-lock.json be reverted.  Mongo-db-memory server is in optional dependencies because its only used for testing. for production, optional dependencies aren't loaded or build, but dev dependencies are still required.   If it didn''t work in your environment, let me know what the problem wass.  Thanks.

Pratik-7331-sjsu commented:
I was getting module not available error while running tests, so I added mongodb-memory-server to dev dependencies instead of just installing it on my machine. Sorry about that. I have reverted the changes. Thanks!

ddfridley commented on app/socket-apis/upsert-why.js:
```diff
@@ -10,6 +15,15 @@ async function upsertWhy(pointObj, cb) {
+  // Joi validation for the category
+  try {
+    await schema.validateAsync({category: pointObj.category});
+    console.log(`Category validation successful for: ${pointObj.category}`);
```
1. In general, please remove console.log statements before final checking. 

2. I know that the example is using validateAsync but can you look into just using validate and getting rid of the try / catch / await.  It's extra overhead that's not adding value here.

Pratik-7331-sjsu replied:
Thank you for the suggestion! I'll update the code to use **validate** instead of validateAsync and remove the try/catch and await to reduce overhead.

ddfridley reviewed (changes requested):
It's all working, but lets clean things up a bit. See comments.  Thanks!

ddfridley reviewed (changes requested):
This is great, and it works, but I missed this last time, please include a comment at the top of each file with a link to the github issue.  It part of the [React Coding style guide](https://github.com/EnCiv/.github/wiki/React-Coding-and-Style-Guidelines) but I guess we need to make a javascript coding guide for what's not React.  Thanks!

ddfridley reviewed (approved):
Looks great! Thanks.
